src/core/border_router/routing_manager.cpp

```diff
@@ -1866,6 +1885,20 @@ void RoutingManager::FavoredOmrPrefix::SetFrom(const OmrPrefix &aOmrPrefix)
     mIsDomainPrefix = aOmrPrefix.IsDomainPrefix();
 }
 
+bool RoutingManager::FavoredOmrPrefix::IsFavoredOver(const OmrPrefix &aOmrPrefix) const
```

Comment:
We now have two `IsFavoredOver()` variants which do slightly different things (which can be confusing):
- The existing one checks preference and then then the smaller prefix (in the sense defined by `Ip6::Prefix`) is favored.
- This one only checks preference.

I understand the reason behind the difference and how they are used in this PR:
- The existing one is used to determine the favored prefix among all prefix entries in netdata.
- The new one is used to tell whether we need to replace the current favored one (from netdata) with the PD prefix (due to it having higher preference)

I suggest:
- We remove the new `IsFavoredOver()`
- Instead in `Evaluate()` we directly check for the case where we have a PD prefix and it has higher preference over the current favored prefix from netdata.